sqlglot/parser.py

```diff
@@ -6485,10 +6491,14 @@ def _parse_alter_table_set(self) -> exp.AlterSet:
 
         return alter_set
 
+    def _parse_alter_view_as(self) -> t.Optional[exp.Expression]:
+        return self._parse_select(self._prev)
+
     def _parse_alter(self) -> exp.AlterTable | exp.Command:
         start = self._prev
 
-        if not self._match(TokenType.TABLE):
+        create_token = self._match_set(self.ALTERABLES) and self._prev
+        if not create_token:
```

Comment:
I think this needs to be changed to `alter_token` or something, there's no `CREATE` in this context.

Reply:
renamed